src/lib/DefaultRuleManager.ts

```diff
@@ -126,6 +133,28 @@ export class DefaultRuleManager implements RuleManager {
 		}
 	}
 
+	protected async emitRunTelemetry(runDescriptorList: ExtendedRunDescriptor[]): Promise<void> {
+		// Telemetry data always specifies an event name.
+		const runTelemetryObject: TelemetryData = {
+			eventName: "ENGINE_EXECUTION"
+		};
+
+		// For engine execution, the telemetry data should include information about which engines ran and which engines
+		// didn't.
+		const executedEngineNames: Set<string> = new Set(runDescriptorList.map(descriptor => descriptor.engine.getName().toLowerCase()));
+		const allEngines: RuleEngine[] = await Controller.getAllEngines();
+		for (const engine of allEngines) {
+			const engineName = engine.getName().toLowerCase();
+			runTelemetryObject[engineName] = executedEngineNames.has(engineName);
```

Comment:
I associated each engine with a boolean property instead of adding an `engines` property with a string, because it seems like the former will be easier to use for meaningful metrics.